scripts/src/public/services/offers.ts

```diff
@@ -37,20 +37,28 @@ export async function getOffers(
 	//  AND app_offers.app_id = ${appId}`
 	// );
 	const dbOffers = await db.Offer.find();
-	const offers: Offer[] = await Promise.all(
-		dbOffers.map(async offer => {
-			const content: db.OfferContent = await offerContents.getOffer(offer.id, logger);
-			return {
-				id: offer.id,
-				title: offer.meta.title,
-				description: offer.meta.description,
-				image: offer.meta.image,
-				amount: offer.amount,
-				blockchain_data: offer.blockchainData,
-				offer_type: offer.type,
-				content: content.content,
-				content_type: content.contentType,
-			};
-		}));
+	const offers = await Promise.all(
```

Comment:
why not `const offers: Offer[] = ...`

Reply:
because the type of the result is `Offer | undefined`, the compiler doesn't know that we're filtering the null values.
so we need to cast it (`as Offer[]`)